### PR TITLE
Et errors

### PR DESCRIPTION
**Pull request by tomekn**
https://trello.com/c/6Kd242iL/187-2-improve-exact-target-er

@cpoo22 @Joachim31415 @dawid-sklodowski

### CODE REVIEW

**Comment by tomekn** [comment]
[(2) Improve Exact Target error handling](https://trello.com/c/6Kd242iL/187-2-improve-exact-target-error-handling)

File: lib/fuelsdk/soap.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ def unpack_body raw
       def unpack raw
         @code = raw.http.code
         unpack_body raw
-        @success = @message == 'OK'
+        @success = raw.hash[:envelope][:body][:create_response][:results][:status_message] == "Created TriggeredSend"
```
Comment (ngsmrk) [inline lib/fuelsdk/soap.rb]: Is this SOAP request class a generic utility used by ET elsewhere - for example when we check if emails are bounced? If so, this change will break that code. See https://github.com/simplybusiness/Skynet/blob/master/lib/skynet/exact_target/bounced_emails_presenter.rb for example.

Comment (dawid-sklodowski) [reply lib/fuelsdk/soap.rb]: Shouldn't condition be a conjunction, ie `@message == 'OK' && status_message == "CreatedTriggeredSend"` ?

Comment (cpoo22) [reply lib/fuelsdk/soap.rb]: Doh! I didn't think of that.

File: spec/triggered_send_response_spec.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    double('raw', hash: data)
+  end
+
+  def build_empty_raw
+    data = { :envelope => { :body => { :create_response => {}}}}
+    double('raw', hash: data)
+  end
+
+  let(:inner_response) { double('FuelSDK::Response') }
+  subject { described_class.new(inner_response) }
```
Comment (cpoo22) [inline spec/triggered_send_response_spec.rb]: Arn't we meant to be using named subjects?

File: spec/triggered_send_response_spec.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    end
+
+    it 'returns true if the SOAP request was successful and TriggeredSend was created' do
+      raw = build_raw 'Created TriggeredSend'
+      expect(inner_response).to receive(:raw).and_return(raw)
+      expect(inner_response).to receive(:success).and_return(true)
+      expect(subject.success).to be true
+    end
+
+    it 'returns false if the SOAP request was successful but the status message wasnt "Created TriggeredSend"' do
```
Comment (cpoo22) [inline spec/triggered_send_response_spec.rb]: Is this not a duplicate of the test at line 31?

File: spec/triggered_send_spec.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    it 'returns a FuelSDK::TriggeredSendResponse' do
+      mock_response = double('FuelSDK::Response')
+      allow(client).to receive(:soap_post).with('TriggeredSend', []).and_return mock_response
+      response = subject.send
+      expect(response).to be_instance_of(FuelSDK::TriggeredSendResponse)
+      expect(response.__getobj__).to eq mock_response
+    end
+  end
+end
```
Comment (cpoo22) [inline spec/triggered_send_spec.rb]: new line 👮

File: spec/triggered_send_spec.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
+require 'spec_helper.rb'
+
+describe FuelSDK::TriggeredSend do
+
+  let(:client) { double('FuelSDK::Client') }
+  subject { described_class.new }
```
Comment (cpoo22) [inline spec/triggered_send_spec.rb]: Same here for named subject

**Pull request merged**